lib/iris/tests/unit/lazy_data/test_convert_nans_array.py

```diff
@@ -0,0 +1,126 @@
+# (C) British Crown Copyright 2017, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Test :func:`iris._lazy data.convert_nans_array` function.
+
+"""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+
+import numpy as np
+import numpy.ma as ma
+
+from iris._lazy_data import convert_nans_array
+
+
+class Test(tests.IrisTest):
+    def setUp(self):
+        self.array = np.array([[1.0, np.nan],
+                               [3.0, 4.0]])
+
+    def test_pass_thru_masked_array_float(self):
+        array = ma.arange(10, dtype=np.float)
+        result = convert_nans_array(array)
+        self.assertIsInstance(result, ma.MaskedArray)
+        self.assertIs(result, array)
+
+    def test_pass_thru_masked_array_integer(self):
```

Comment:
***Un*** -masked ints are pass-through anyway.
Should be checking that.
It's in the docstring!